workshops/platformer/README.md

```diff
@@ -0,0 +1,717 @@
+
+This is because at the very beginning of the game, no obstacles have yet been generated! Our `obstacleSprites` is empty, and there are no obstacles to check the position of! To solve this, we must add the condition `obstacleSprites.length > 0` to our conditional:
+
+Here we're saying "if obstacleSprites has more than 0 things, AND the right edge (offsetting!) of the first obstacle is less than the left-screen bound, then get rid of it."
+
+```js
+if (obstacleSprites.length > 0 && firstObstacle.position.x <= camera.position.x - (width/2 + firstObstacle.width/2)) {
+  removeSprite(firstObstacle);
+}
+```
+
+### Detecting Collisions
+
+Great, we've got a good assortment of randomly generated obstacles. Let's add collisions!
+
+p5.play has an [`overlap`]() method for Groups that makes collision detection and handling easy. We can use it on our `obstacleSprites` Group by providing two arguments, the thing to collide with, and a function to do something when there is a collision.
+
+Add this to the `draw()` function, right before the call for `drawSprites()`:
+
+```js
+obstacleSprites.overlap(player, endGame);
+```
+
+In this line, we are handling a collision between any of the sprites in `obstacleSprites` and our player. When a collision happens, we'll run the `endGame` function (which we'll define next).
+
+### Handling Collisions
+
+At the bottom of `game.js`, below our `draw()` function, we can add our `endGame` function:
+
+```js
+function endGame() {
+
+}
+```
+
+Let's use something called `console.log()` to create output in our console. If we add this to `endGame()`, we'll see "Game Over!" in our console in the Inspector.
+
+```js
+function endGame() {
+  console.log("Game Over!");
+}
+```
+
+Save and view your game in the external live preview. Check your console in the Inspector.
+
+Great. Now that we've made sure that collisions get handled properly by this function, we can flesh out the game-over behavior a bit more.
+
+## Part V: Ending Game
+
+Similar to what we did in Dodge, we'll be splitting the `draw()` function into two modes: game over and game not over. Let's add a variable to keep track of this at the top of the file, and initialize it as `false` at the top of `setup()`.
+
+```js
+var isGameOver;
+
+function setup() {
+  isGameOver = false;
+}
+```
+
+We'll add a line to set this flag to true within our `endGame()`:
+
+```js
+function endGame() {
+  isGameOver = true;
+  console.log("Game Over!");
+}
+```
+
+And we'll create the two modes now:
+
+- add a conditional at the top for the game over mode
+
+  ```js
+  function draw() {
+    if (isGameOver) {
+
+    }
+
+    background(150, 200, 250);
+
+    ...
+  }
+  ```
+
+- wrap the game logic we've written in `draw()` so far in `else { }`.
+
+  ```js
+  function draw() {
+    if (isGameOver) {
```

Comment:
clarify to put specific functions such as gameOver in the if conditional, and the else conditional contains the original game code we made previously.